src/main/java/games/strategy/triplea/attachments/TechAbilityAttachment.java

```diff
@@ -49,21 +49,19 @@ public static TechAbilityAttachment get(final TechAdvance type) {
         return hardCodedTechAttachment;
       }
     }
-    final TechAbilityAttachment rVal =
-        (TechAbilityAttachment) type.getAttachment(Constants.TECH_ABILITY_ATTACHMENT_NAME);
-    return rVal;
+    return (TechAbilityAttachment) type.getAttachment(Constants.TECH_ABILITY_ATTACHMENT_NAME);
   }
 
   /**
    * Convenience method.
    */
   public static TechAbilityAttachment get(final TechAdvance type, final String nameOfAttachment) {
-    final TechAbilityAttachment rVal = (TechAbilityAttachment) type.getAttachment(nameOfAttachment);
-    if (rVal == null) {
+    final TechAbilityAttachment techAbilityAttachment = (TechAbilityAttachment) type.getAttachment(nameOfAttachment);
+    if (techAbilityAttachment == null) {
       throw new IllegalStateException(
           "No technology attachment for:" + type.getName() + " with name:" + nameOfAttachment);
     }
-    return rVal;
+    return techAbilityAttachment;
```

Comment:
Those getAttachment methods are all pretty similar, they could be unified and just call another method with a custom error message